src/gConsent/manage/approveAccessRequest.test.ts

```diff
@@ -64,47 +64,61 @@ jest.mock("cross-fetch");
 const mockAcpClient = (
   options?: Partial<{
     hasAccessibleAcr: boolean;
-    initialResource: unknown;
-    updatedResource: unknown;
+    initialResource: SolidClient.WithServerResourceInfo & SolidClient.WithAcp;
+    updatedResource: SolidClient.WithAccessibleAcr;
   }>
 ) => {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const solidClientModule = jest.requireMock("@inrupt/solid-client") as any;
+  const solidClientModule = jest.requireMock(
+    "@inrupt/solid-client"
+  ) as jest.Mocked<typeof SolidClient>;
   solidClientModule.acp_ess_2.hasAccessibleAcr.mockReturnValueOnce(
     options?.hasAccessibleAcr ?? true
   );
 
-  solidClientModule.acp_ess_2.setVcAccess.mockReturnValueOnce(
-    options?.updatedResource ?? {}
-  );
-  solidClientModule.acp_ess_2.getResourceInfoWithAcr.mockResolvedValueOnce(
-    options?.initialResource ?? {}
-  );
-  solidClientModule.acp_ess_2.saveAcrFor = jest.fn();
+  if (options?.updatedResource) {
+    solidClientModule.acp_ess_2.setVcAccess.mockReturnValueOnce(
+      options?.updatedResource
+    );
+  }
+
+  if (options?.initialResource) {
+    solidClientModule.acp_ess_2.getResourceInfoWithAcr.mockResolvedValueOnce(
+      options?.initialResource
+    );
+  }
+
+  solidClientModule.acp_ess_2.saveAcrFor = jest.fn<
+    typeof SolidClient.acp_ess_2.saveAcrFor
+  >() as any;
   return solidClientModule;
 };
 
+const mockedInitialResource = mockSolidDatasetFrom(
```

Comment:
@NSeydoux what do you think about making `mockSolidDatasetFrom` accept a generic? So:

```
const mockedInitialResource = mockSolidDatasetFrom<
  SolidClient.WithServerResourceInfo & SolidClient.WithAcp
>(
"https://some.resource"
);
```

This might be better than casting to unknown

Reply:
I would prefer not to: this cast is quite hacky, and it's only okay because it just ends up satisfying the mock defined here: 
```
solidClientModule.acp_ess_2.getResourceInfoWithAcr.mockResolvedValueOnce(
      options?.initialResource
    );
``` 
Arguably, it would be possible to add the cast to unknown there, rather than when initialling the variable. In any case, that's a dirty cast, and I'd rather keep the mock object have an accurate type, rather than picking it up from the generic, since `mockSolidDatasetFrom` is part of the public API, so I wouldn't want it to allow to do things that are too crazy.

Reply:
hmm, yeah, let's do the casting at the call site then, I guess? idk. Minor issue tbh.

Reply:
Maybe we'd more want something like `mockSolidDatasetFrom(resourceIri: string, { withAcr: string / ACR object })` or something?